contributing.md

```diff
@@ -0,0 +1,43 @@
+
+## Submitting a bug report
+
+If you encounter a bug, please verify that the bug still exists on the latest version of Pylake. You can find instructions on how to update Pylake [here](https://lumicks-pylake.readthedocs.io/en/stable/install.html#updating).
+
+If the bug persists, please open an issue describing the bug. This issue should at least contain the following information:
+
+- A short summary of the bug. In most cases, this should only be a couple of sentences.
+- A code snippet to reproduce the bug. Please do your best to reduce the code snippet to the minimum required.
+- The result of the code snippet.
+- The expected result of the code snippet (which is different from the obtained result because of the bug).
+
+Please also make sure you mention the version of Python, the platform and the version of Pylake. If reproducing the bug relies on data, please make sure this data is accessible to us.
+
+## Preparing a Pull Request (PR)
+
+Code and documentation contributions will generally proceed through a Pull Request.
+
+- Please provide a clear PR description detailing the change in functionality the PR provides. If the implemented method is based on a paper, please include a reference to the paper.
+- If the PR is related to an open issue, please refer to this issue in the PR description.
+- Try and keep the pull request manageable in terms of size. If a new feature is complex and requires multiple steps, consider breaking it up into multiple Pull Requests for the different sub-problems. If you struggle with this, please open an issue beforehand, and we can try and help you.
+- Keep a clean commit history. This will greatly aid in getting your PR reviewed quickly.
+- New features or changes which are user facing require a changelog entry detailing these changes. They also require documentation describing their use and function. Documentation can be found in the `docs` directory. This directory also contains instructions on how to locally build the docs such that you can verify everything renders correctly.
+- New features must include unit tests which ensure their correct function. Tests for new functionality should ideally have 100% coverage. This will ensure that future development or dependency updates will not accidentally break your feature. To measure coverage, you can use the package [pytest-cov](https://pypi.org/project/pytest-cov/).
+- Make sure all tests pass after you make your changes.
+- You may merge the Pull Request once you have a sign-off of one core developer, or if you do not have permission to do that, you may request this developer to merge it for you. We may request you to rebase your PR to clean up the commit history (and rebase to the main branch). If you feel uncomfortable with this, please reach out to us for help. Note that using a graphical git client such as [fork](https://git-fork.com/) or [GitKraken](https://www.gitkraken.com/) can greatly simplify this process.
```

Comment:
I would leave out the sentence 'if you feel uncomfortable..' A user that got this far can probably do the merging too.

Reply:
True!